### PR TITLE
Register LinkFactory in MEF

### LINKED ISSUE #10219: [NBug] Index was outside the bounds of the array.
### Environment

- Git Extensions 4.0.0.15291
- Build 8b80e54f15e8923a85f5801cca033f932ced821f
- Git 2.37.3.windows.1
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.9
- DPI 96dpi (no scaling)

### Issue description

```
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Microsoft.VisualStudio.Composition.Reflection.ResolverExtensions.Resolve(ParameterRef parameterRef)
   at Microsoft.VisualStudio.Composition.Reflection.ParameterRef.get_ParameterInfo()
   at Microsoft.VisualStudio.Composition.ImportDefinitionBinding.get_ImportingParameter()
   at Microsoft.VisualStudio.Composition.ComposedPart.GetDiagnosticLocation(ImportDefinitionBinding import)
   at Microsoft.VisualStudio.Composition.ComposedPart.Validate(IReadOnlyDictionary`2 metadataViews)+MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at Microsoft.VisualStudio.Composition.CompositionConfiguration.Create(ComposableCatalog catalog)
   at GitUIPluginInterfaces.ManagedExtensibility.CreateExportProvider(String applicationDataFolder)
   at GitUIPluginInterfaces.ManagedExtensibility.<>c__DisplayClass7_0.<GetOrCreateLazyExportProvider>b__0()
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at GitUIPluginInterfaces.ManagedExtensibility.GetExport[T]()
   at GitUI.CommitInfo.CommitInfo..ctor()
   at GitUI.CommandsDialogs.FormBrowse.InitializeComponent()
   at GitUI.CommandsDialogs.FormBrowse..ctor(GitUICommands commands, BrowseArguments args)
   at GitUI.GitUICommands.StartBrowseDialog(IWin32Window owner, BrowseArguments args)
   at GitExtensions.Program.RunApplication()
   at GitExtensions.Program.Main()
```

Got this error starting GitExtensions.

### Steps to reproduce

I got this error after installing the v4 beta version and trying to start it.

### Did this work in previous version of GitExtensions?

_No response_

### Diagnostics

_No response_

### PR DESCRIPTION
**Pull request by mstv**
Fixes #10219

## Proposed changes

- Register the actually needed type `LinkFactory` in the MEF

## Screenshots 

N/A

## Test methodology 

- manual

## Test environment(s) 

- Git Extensions 33.33.33
- Build 7d3489c42c1c4db12b35bd1acd07741a022bac0a
- Git 2.35.2.windows.1 (recommended: 2.37.1 or later)
- Microsoft Windows NT 10.0.19044.0
- .NET 6.0.9
- DPI 96dpi (no scaling)

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitExtensions/Program.cs [hunk 1]
```diff
@@ -110,7 +110,7 @@ private static void RunApplication()
                 {
                     typeof(GitUI.GitExtensionsForm).Assembly,
                     typeof(GitCommands.GitModule).Assembly,
-                    typeof(ResourceManager.GitPluginBase).Assembly
+                    typeof(ResourceManager.LinkFactory).Assembly
```
Comment (RussKie) [inline GitExtensions/Program.cs:113]: This change is essentially a no-op - we are registering an assembly from which MEF is loading exports.
Does LinkFactory need to be marked as Export?

**Comment by RussKie** [comment]
Were you able to reproduce the crash?

**Review by gerhardol — COMMENTED** [review]
No info

File: GitExtensions/Program.cs (hunk 1 above)
Comment (mstv) [reply GitExtensions/Program.cs:113]: > This change is essentially a no-op - we are registering an assembly from which MEF is loading exports.

Yes, that's why "wild guess".

> Does LinkFactory need to be marked as Export?

It *is* exported.

**Comment by mstv** [comment]
> Were you able to reproduce the crash?

No, it has always worked for me.

**Pull request closed without merging**